src/main/java/jenkins/metrics/api/Metrics.java

```diff
@@ -451,6 +441,11 @@ public void run() {
                             }
                         } catch (InterruptedException e) {
                             e.printStackTrace(l.fatalError("aborted"));
+                        } catch (Exception e) {
+                            logger.log(Level.SEVERE, "Error running " + HealthChecker.class.getName(), e);
+                            if (l != null) {
+                                e.printStackTrace(l.fatalError(e.getMessage()));
+                            }
```

Comment:
the logger goes to a file and gets rewritten every time (in my experience, don't know why), and it took me a while to figure out what was wrong. There no thing as too much logging!